### ISSUE TITLE
secret value is empty after upgrade in 1.3.0

### ISSUE THREAD

**Issue by webedia-it** [hashicorp/vault#8006]
**Describe the bug**
After upgrade from 1.2.4 to 1.3.0 I have some secrets that appears empty in vault gui.
For information I have tried with 1.2.4 I don't have this problem

**To Reproduce**
1. add a secret with on multiple lines (with "\n" in json)
2. upgrade your vault in 1.3.0
3. go in http://host:8200/ui/vault/secrets/
4. value is empty after you click on eye button
5. if you check in json you will see the value content

**Expected behavior**
See values with "\n" after upgrade in 1.3.0

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.2.4
* Vault CLI Version (retrieve with `vault version`): 1.2.4
* Server Operating System/Architecture: Debian 9.11

Vault server configuration file(s):

```hcl
storage "consul" {
  address = "127.0.0.1:8500"
  path    = "vault"
  token   = "secret"
}

listener "tcp" {
  address     = "0.0.0.0:8200"
  cluster_address  = "192.168.0.1:8201"
  tls_disable = "true"
}

ui = true
default_lease_ttl = "360h"
api_addr = "http://192.168.0.1:8200"
cluster_addr = "https://192.168.0.1:8201"

```

**Additional context**

**Comment by gondaz** [comment]
I confirm the same behavior for me after upgrade to 1.3.0 (etcd storage backend). Moreover, using this version (1.3.0) I create a new secret, and then revealing the secret by clicking on the eye button I see an empty string, but it happens occasionally, not every time.

**Comment by michelvocks** [comment]
Hi @webedia-it & @gondaz!

I think this issue is a duplicate of https://github.com/hashicorp/vault/pull/7926.
Could you please confirm that this issue only occurs when the secret has a dot "." included?

Cheers,
Michel

**Comment by gondaz** [comment]
Hi @michelvocks !
Thanks for the quick response. I checked that on my environment and I can confidently confirm the bug appears only when there is a dot among the key field.

Will that mentioned PR/fix be included in the next Vault release?

**Comment by michelvocks** [comment]
> Will that mentioned PR/fix be included in the next Vault release?

Yes. It will be available with Vault 1.3.1.

**Comment by webedia-it** [comment]
Hi @michelvocks I confirm it's a duplicate issue. I close this one.
Thank you
